src/dataflow/components/nodes/factories/data-storage-rete-node-factory.tsx

```diff
@@ -0,0 +1,111 @@
+import Rete from "rete";
+import { Node, Socket } from "rete";
+import { NodeData } from "rete/types/core/data";
+import { DataflowReteNodeFactory } from "./dataflow-rete-node-factory";
+import { NumControl } from "../controls/num-control";
+import { TextControl } from "../controls/text-control";
+import { PlotControl } from "../controls/plot-control";
+import * as uuid from "uuid/v4";
+
+export class DataStorageReteNodeFactory extends DataflowReteNodeFactory {
+  constructor(numSocket: Socket) {
+    super("Data Storage", numSocket);
+  }
+
+  public builder(node: Node) {
+    const inputUuid = uuid();
+    const inp = new Rete.Input("i-" + inputUuid, "sequence", this.numSocket);
+    inp.addControl(new TextControl(this.editor, "it-" + inputUuid, node, "", "my-sequence"));
```

Comment:
Open to feedback on whether this is too aggressive.  In previous nodes where we have static inputs, inputs are given keys like "num1", "num2" etc.  In this case we have variable inputs and inputs can be removed from any position.  Rather than trying to reshuffle inputs to keep the ordered numeric key names, I went with a unique UUID for each input key.  This simplifies input deletion (since name isn't tied to order - any input can be arbitrarily deleted without worrying about reorganizing remaining inputs).  Not sure if this is too heavy-handed though.  Definitely different than other Rete keys which have very simple names.  Thoughts are welcome.

Reply:
Agreed.  I like having the input keys for the data storage node look more like the input keys for other nodes.

Reply:
Updated to use a simple counter.